app/src/main/java/com/arctouch/codechallenge/util/PaginationScrollListener.kt

```diff
@@ -0,0 +1,29 @@
+package com.arctouch.codechallenge.util
+
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+
+abstract class PaginationScrollListener(internal var layoutManager: LinearLayoutManager) : RecyclerView.OnScrollListener() {
```

Comment:
You encapsulated the scroll logic inside a separated class. Good job! 👏

Reply:
Thank you! ^^